src/Features/CSharp/Portable/Completion/CompletionProviders/FileBasedPrograms/PropertyAppDirectiveCompletionProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+
+[ExportCompletionProvider(nameof(PropertyAppDirectiveCompletionProvider), LanguageNames.CSharp)]
+[ExtensionOrder(After = nameof(SdkAppDirectiveCompletionProvider))]
+[Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class PropertyAppDirectiveCompletionProvider() : AbstractAppDirectiveCompletionProvider
+{
+    protected override string DirectiveKind => "property";
+
+    protected sealed override void AddDirectiveKindCompletion(CompletionContext context)
+    {
+        context.AddItem(CommonCompletionItem.Create(DirectiveKind, displayTextSuffix: "", CompletionItemRules.Default, glyph: Glyph.Keyword,
+            description: [
+                new(SymbolDisplayPartKind.Keyword, symbol: null, "#:property"),
```

Comment:
Where is `SymbolDisplayPartKind.PreprocessorText`? I guess Keyword is close. Is there plans on classifying these differently than PreprocessorText?

Reply:
No objection to changing this, I just used it because I chose the "keyword" glyph for the completion item corresponding to this.

Also, it looks like this formatting might not be respected in Quick Info in vscode, or, I am just doing something wrong. Since it actually appears as plain text for me, instead of being classified the way that it is in source code.

Reply:
https://github.com/dotnet/csharp-tmLanguage/tree/main It would be interesting to see how the TM grammar is classifying these. I would only make it work for being in source text. I have joked with @dibarbet that we could handle QuickInfo by encoding a series of zero-width spaces to them...

Reply:
dotnet/csharp-tmLanguage#337